### PR TITLE
Openzfs restructure part 2

### PR DESCRIPTION
**Pull request by mattmacy**
This is the second part of an effort to break out all the non-functional changes from 8987.
Reviewers should look at the second commit. 
The first commit is https://github.com/zfsonlinux/zfs/pull/9198 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by lundman** [comment]
I notice Linux still only uses `Makefile.in` for the kernel parts, whereas OSX does `Makefile.am` the whole way - but I'm sure we can find a way around that, sometime in the far distant future.

**Comment by mattmacy** [comment]
> I notice Linux still only uses `Makefile.in` for the kernel parts, whereas OSX does `Makefile.am` the whole way - but I'm sure we can find a way around that, sometime in the far distant future.

Linux and FreeBSD both rely on their own respective build infrastructure.

File: module/zfs/spa_misc.c [hunk 1]
```diff
@@ -59,7 +59,9 @@
 #include <sys/kstat.h>
 #include "zfs_prop.h"
 #include <sys/zfeature.h>
-#include "qat.h"
+#ifdef __linux__
```
Comment (behlendorf) [inline module/zfs/spa_misc.c]: Can we drop the `__linux__` wrapper here and make sure proper compatibility code is provided for the other platforms.

**Comment by behlendorf** [comment]
@lundman @ikozhukhov the header restructuring has been integrated, would you mind looking over this next chunk which follows that up by relocating the platform specific sources.

**Comment by ikozhukhov** [comment]
probably we need rename `module/zfs/Makefile.in` to `module/zfs/Makefile.am` for consistency with others

**Comment by behlendorf** [comment]
> probably we need rename Makefile.in to Makefile.am for consistency with others

Definitely a possibility, though to be clear the primary intent of this PR is only to restructure the existing sources to split off the platform specific files.  That will make it easier to follow up with additional portability changes.

File: module/Makefile.in [hunk 2]
```diff
@@ -27,25 +27,38 @@ export ZFS_MODULE_CFLAGS ZFS_MODULE_CPPFLAGS
...
 
 	find . -name '*.ur-safe' -type f -print | xargs $(RM)
 
-modules_install:
+clean: clean-@ac_system@
+
+modules_install-Linux:
 	@# Install the kernel modules
-	$(MAKE) -C @LINUX_OBJ@ M=`pwd` $@ \
+	$(MAKE) -C @LINUX_OBJ@ M=`pwd` modules_install \
```
Comment (behlendorf) [inline module/Makefile.in]: The updated install target results in the `spl.ko` file being install in a slightly different location.  This works in my initial testing, but could cause problems for end users or any scripts expecting to find them is the previous location.  Let's tweak this to preserve the original location.

```
./extra/avl/zavl.ko
./extra/icp/icp.ko
./extra/lua/zlua.ko
./extra/nvpair/znvpair.ko
./extra/os/linux/spl/spl.ko (used to be installed as /extra/spl/spl.ko)
./extra/unicode/zunicode.ko
./extra/zcommon/zcommon.ko
./extra/zfs/zfs.ko
```

Comment (mattmacy) [reply module/Makefile.in]: It's just basing it off the source path. How do I special case one directory?

Comment (mattmacy) [reply module/Makefile.in]: I don't see any way to do that without either changing everything to point to `module/spl`for objects but keep sources in `module/os/linux/spl` - which would be super messy. I'd rather not unless this is known to actually break something.

Comment (mattmacy) [reply module/Makefile.in]: Never mind. I changed the build to build spl.ko in `module/spl`. :man_shrugging:

Comment (behlendorf) [reply module/Makefile.in]: My specific concern was the changing the location could lead to problems when upgrading/downgrading the packages.   Thanks, for fixing this!

**Comment by behlendorf** [comment]
Looks good there were 3 ZTS failures but all for known spurious issues.

**Pull request merged**